### ISSUE TITLE
Absent gradZ after update (3-D segmentation; no cells segmented)

### ISSUE THREAD

**Issue by b2jia** [MouseLand/cellpose#145]
I recently updated ```cellpose``` only to find that after training my own model, it outputs no segmentation masks. ```gradXY``` (formerly flowXY), ```cellprob``` are all non-zero, but ```gradZ``` is noticeably absent:

```image```
![image](https://user-images.githubusercontent.com/37123556/98633477-5804fe00-22d6-11eb-8f16-c1f9c54385a1.png)
```gradXY```:
![image](https://user-images.githubusercontent.com/37123556/98633485-5d624880-22d6-11eb-981e-c8f5f1c8d75c.png)
```cellprob```:
![image](https://user-images.githubusercontent.com/37123556/98633603-96022200-22d6-11eb-9bb1-8567a645d0e2.png)
```gradZ```:
![image](https://user-images.githubusercontent.com/37123556/98633568-897dc980-22d6-11eb-9f19-fd38f341b108.png)

**Comment by carsen-stringer** [comment]
is this the github version or the pip? I can't reproduce this with the github version

**Comment by carsen-stringer** [comment]
oh I see you said your own model -- I don't see why that would have changed though, I've added an anisotropy parameter too

**Comment by b2jia** [comment]
This was the pip version. After updating to 0.1.0.1 I am experiencing this error. Here is how I trained:

```python -m cellpose --train --dir 2D/ --img_filter img --mask_filter masks --use_gpu --pretrained_model cyto --chan 0 --chan2 0 --diameter 30 --n_epochs 2000```

and here is how I evaluated with my trained model:

```
# Define cellpose model

model = models.CellposeModel()
model = cellpose.models.CellposeModel(gpu=True,
                                      diam_mean = 30.0,
                                      net_avg = True,
                                      pretrained_model = 'cellpose_train/D2/2D/models/cyto_cellpose_residual_on_style_on_concatenation_off_2D_2020_11_09_15_13_03.125916',
                             )

for i in range(0, 27):
    
        idx = '%02d' % i
        
        input_vol= tifffile.imread('Exp2/DAPI/cellpose_retrained/HCT_{}_orig.tif'.format(idx))

        # define channels
        channels = [[0,0]]

        masks, flows, styles = model.eval([input_vol], 
                                          channels=channels, 
                                          do_3D=True,
                                          net_avg = True,
                                          invert=False)

        # save results to correct in gui
        io.masks_flows_to_seg([input_vol], 
                              masks, 
                              flows, 
                              [32], 
                              ['Exp2/DAPI/cellpose_retrained/HCT_{}_orig.tif'.format(idx)])
        
        # clear GPU cache
        mx.gpu(0).empty_cache()
```

If it helps, I'm happy to upload the model weights to help reproduce this error? I admit I'm a little flustered about not having any lead on where this bug is.

**Comment by b2jia** [comment]
[cyto_cellpose_False_2D_2020_11_09_12_31_13.zip](https://github.com/MouseLand/cellpose/files/5522435/cyto_cellpose_False_2D_2020_11_09_12_31_13.zip)

**Comment by carsen-stringer** [comment]
I loaded and ran your model but got a gradZ, see figure below:
![image](https://user-images.githubusercontent.com/15067541/100158909-28273000-2e6a-11eb-8f2e-abfc00696881.png)

I ran this on 3D nuclear data, it doesn't seem like the model was trained on enough data, or maybe it was trained on data that's very different from nuclei.
